pkgs/by-name/ba/balatro/package.nix

```
@@ -0,0 +1,79 @@
+{
+  stdenv,
+  lib,
+  pkgs,
+  makeDesktopItem,
+  requireFile,
+  withMods ? true,
+}:
+let
+  version = "1.0.1g";
+  balatroExe = requireFile {
+    name = "Balatro-${version}.exe";
+    url = "https://store.steampowered.com/app/2379780/Balatro/";
+    # Use `nix hash file --sri --type sha256` to get the correct hash
+    hash = "sha256-X7t/T3lkLnDzewlmkxHjdePH/hlwoachuWfoUC4ntd0=";
+  };
+in
+stdenv.mkDerivation {
+  pname = "balatro";
+  inherit version;
+  nativeBuildInputs = with pkgs; [
+    p7zip
+    copyDesktopItems
+  ];
+  buildInputs = with pkgs; [ love ] ++ lib.optional withMods pkgs.lovely-injector;
+  src = ./.;
+  desktopItems = [
+    (makeDesktopItem {
+      name = "balatro";
+      desktopName = "Balatro";
+      exec = "balatro";
+      keywords = [ "Game" ];
+      categories = [ "Game" ];
+      icon = "balatro";
+    })
+  ];
+  buildPhase = ''
+    runHook preBuild
+    mkdir -p $out/share/{applications,icons}
+
+    # "Patch" game and extract icon
+    tmpdir=$(mktemp -d)
+    7z x ${balatroExe} -o$tmpdir -y
+    patch $tmpdir/globals.lua < $src/globals.patch
```

```suggestion
    patch $tmpdir/globals.lua -i ${./globals.patch}
```

also this needs a comment to explain why we're doing this

---

The patch is needed so that the 'Linux' platform behave the same as the other Desktops.
The patch does the following:
* Set the automatic save timer to a lower value compared to the standard one
* Enables the discord button on the main screen
* Enables the language options in the main screen

Looking again at it, I noticed that the `CRASH_REPORTS` and the `NO_ACHIEVEMENTS` variables could be omitted from the patch, since CRASH_REPORTS is false by default and NO_ACHIEVEMENTS works well since Steam (or another achievement system) is not present in this installation (thus showing an in-game animation for the achievement)